### DISCUSSION TITLE
An option to blend hull spaces given the correct conditions.

### DISCUSSION THREAD

**Discussion by Xeavas** [FakeFishGames/Barotrauma#5502] [Ideas and suggestions]
Said requirements should prevent most cases of improper use of hull blending

1. If the hulls are linked
2. if the hulls are connected with a permanent open gap
3. if the checkbox to blend hulls is enabled

This would enable sub designers to create more complex hull areas without making the display in the status monitor look out of place or cluttered, introducing a way to "smooth" out hull areas without actually making a triangle hull shape for example.

It would also enable seeing the linked areas as "more complete" as it displays as one rather than highlighting all of them too look as one, status like water % and oxygen % would also be averaged out basically unifying all hull spaces to display as a single hull space. 

Another potential issue to tackle is flow rate within the hull group, it would be made that gaps relating to the hull group have unlimited flow rate within said hull group thus making it look like its functioning as a single hull. Other than that this could be disregarded as a problem for the submarine designer to solve and offer said option as a purely cosmetic feature for the status monitor.

This option would be a compromise to polygonal hull spaces by having it look as close to it in the status monitor but functionally identical to the current hull shapes/mechanics (especially using the bread slicing technique to create "shaped" hulls) 

So with the conditions met, instead of making the status monitor of the submarine looking like this (Nova VX in this example)
![teststatdraw](https://user-images.githubusercontent.com/72674994/114331478-a6b10f80-9b76-11eb-9e47-eb55642c655a.jpg)

It would attempt to blend the hull areas and make it look more like this
![novaeditedstatus](https://user-images.githubusercontent.com/72674994/114331498-b3356800-9b76-11eb-9bde-84fdb72ef32f.jpg)

**Comment by Vectricity** [comment]
While not at all necessary, it would certainly be nice. I support this.
